clients/include/blas2/testing_spr.hpp

```diff
@@ -70,50 +71,67 @@ void testing_spr_bad_arg(const Arguments& arg)
-                              HIPBLAS_STATUS_INVALID_VALUE);
+        DAPI_EXPECT(
+            HIPBLAS_STATUS_NOT_INITIALIZED, hipblasSprFn, (nullptr, uplo, N, alpha, dx, incx, dA));
+        DAPI_EXPECT(HIPBLAS_STATUS_INVALID_VALUE,
+                    hipblasSprFn,
+                    (handle, HIPBLAS_FILL_MODE_FULL, N, alpha, dx, incx, dA));
         EXPECT_HIPBLAS_STATUS(
             hipblasSprFn(handle, (hipblasFillMode_t)HIPBLAS_OP_N, N, alpha, dx, incx, dA),
             HIPBLAS_STATUS_INVALID_ENUM);
 
         if(arg.bad_arg_all)
         {
-            EXPECT_HIPBLAS_STATUS(hipblasSprFn(handle, uplo, N, nullptr, dx, incx, dA),
-                                  HIPBLAS_STATUS_INVALID_VALUE);
+            DAPI_EXPECT(HIPBLAS_STATUS_INVALID_VALUE,
+                        hipblasSprFn,
+                        (handle, uplo, N, nullptr, dx, incx, dA));
 
             if(pointer_mode == HIPBLAS_POINTER_MODE_HOST)
             {
                 // For device mode in rocBLAS we don't have checks for dA, dx as we may be able to quick return
-                EXPECT_HIPBLAS_STATUS(hipblasSprFn(handle, uplo, N, alpha, nullptr, incx, dA),
-                                      HIPBLAS_STATUS_INVALID_VALUE);
-                EXPECT_HIPBLAS_STATUS(hipblasSprFn(handle, uplo, N, alpha, dx, incx, nullptr),
-                                      HIPBLAS_STATUS_INVALID_VALUE);
+                DAPI_EXPECT(HIPBLAS_STATUS_INVALID_VALUE,
+                            hipblasSprFn,
+                            (handle, uplo, N, alpha, nullptr, incx, dA));
+                DAPI_EXPECT(HIPBLAS_STATUS_INVALID_VALUE,
+                            hipblasSprFn,
+                            (handle, uplo, N, alpha, dx, incx, nullptr));
             }
 
             // With alpha == 0, can have all nullptrs
-            CHECK_HIPBLAS_ERROR(hipblasSprFn(handle, uplo, N, zero, nullptr, incx, nullptr));
+            DAPI_CHECK(hipblasSprFn, (handle, uplo, N, zero, nullptr, incx, nullptr));
         }
 
         // With N == 0, can have all nullptrs
-        CHECK_HIPBLAS_ERROR(hipblasSprFn(handle, uplo, 0, nullptr, nullptr, incx, nullptr));
+        DAPI_CHECK(hipblasSprFn, (handle, uplo, 0, nullptr, nullptr, incx, nullptr));
+    }
+
+    if(arg.bad_arg_all && arg.api & c_API_64)
+    {
+        hipblasLocalHandle handle(arg);
+        CHECK_HIPBLAS_ERROR(hipblasSetPointerMode(handle, HIPBLAS_POINTER_MODE_HOST));
+
+        int64_t n_64 = 2147483648; // will rollover to -2147483648 if using 32-bit interface
+        // rocBLAS implementation has alpha == 0 quick return after arg checks, so if we're using 32-bit params,
+        // this should fail with invalid-value
+        // Note that this strategy can't check incx as rocBLAS supports negative. Also depends on implementation so not testing cuBLAS for now
+        DAPI_CHECK(hipblasSprFn, (handle, uplo, n_64, &h_zero, nullptr, 1, nullptr));
```

Comment:
Looks like good approach.  Add contants to yaml is better than magic.   You can still test on 32bit but use DAPI_EXPECT with conditional result, invalid_size : success to prove your test process itself and the constant.